\node at(0,5) {\Huge Soma};

\node[default, draw](soma){Soma};
\question{below left}{1}{soma}{cytoplasm}{cytoplas}{Cytoplasm}
\node[default, draw, below right=of soma, fill=gray](organelle2){Organelle};
\node[default, draw, below=of cytoplasm, fill=gray](organelle1){Organelle};
\node[default, draw, below=of organelle2](nucleus){Nucleus};
\draw[brown, dashed]($(nucleus.north)+(0,.5)$)
-| node[right, switch ocg={nuclearEnvelope nuclearEnvelopee}](nuclearEnvelope){
        \basic{nuclearEnvelope}{Nuclear Envelope}
    }
($(nucleus.east)+(.5,0)$)
|- ($(nucleus.south)+(0,-.5)$)
-| ($(nucleus.west)+(-.5,0)$)
|- cycle;
\node at(nuclearEnvelope){
    \basicr{nuclearEnvelopee}{?}
};
\node[default, draw, above right=of nuclearEnvelope](pore){Pores};
\draw[arrow, green](nuclearEnvelope) to (pore);
\node[below right=of nucleus, yshift=.3cm](space1){};
\question{left}{1}{organelle1}{golgiApparatus}{ga}{Golgi Apparatus}
\node[default, draw, above=.3 of golgiApparatus](mitochondria){Mitochondria};
\question{below}{.3}{golgiApparatus}{endoplasmicReticulum}{pr}{Endoplasmic Reticulum}
\question{above}{.3}{mitochondria}{ribosome}{ribosomee}{Ribosome}
\node[default, draw, above=of ribosome](proteinSynthesis){Protein Sysnthesis};
\node[default, draw, below left=1cm of endoplasmicReticulum, xshift=2cm, switch ocg={rpr rprr}](roughEndoplasmicReticulum){
    \basic{rpr}{Rough Endoplasmic Reticulum}
} node at (roughEndoplasmicReticulum){
    \basicr{rprr}{?}
};
\node[default, draw, below right=1cm of endoplasmicReticulum, xshift=-2cm, switch ocg={spr sprr}](smoothEndoplasmicReticulum){
    \basic{spr}{Smooth Endoplasmic Reticulum}}
node at(smoothEndoplasmicReticulum){
    \basicr{sprr}{?}
};
\node[default, draw, below=of nucleus](chromosome){Chromosome};
\node[right=of chromosome, switch ocg={numb numbb}](46chromosomes){
    \basic{numb}{\textcolor{black}{Number of human's chromosomes:} 46}}
node[right=of chromosome]{\basicr{numbb}{\textcolor{black}{Number of human's chromosomes:} ?}
};
\question{below}{1}{chromosome}{deoxyribonucleicAcid}{da}{Deoxyribonucleic Acid}
\node[right=of deoxyribonucleicAcid, switch ocg={nano nanoo}](2nm){
    \basic{nano}{\textcolor{black}{Same in all the cells / size:} 2$\mu$m}
}
node[right=of deoxyribonucleicAcid]{
    \basicr{nanoo}{\textcolor{black}{Same in all the cells / size:} ?}
};
\node[below right=of deoxyribonucleicAcid, yshift=1cm, switch ocg={ge gee}](geneexpression){
    \basic{ge}{\textcolor{black}{What DNA reading is known as:} Gene Expression}
}
node[below right=of deoxyribonucleicAcid, yshift=1cm]{
    \basicr{gee}{\textcolor{black}{What DNA reading is known as:} ?}
};
\node[default, draw, below=of deoxyribonucleicAcid](gene){Gene};
\node[default, draw, right=5 of gene](rna){RNA};
\question{right}{5}{rna}{spliceosome}{spli}{Spliceosome}
\question{right}{5}{spliceosome}{mrna}{mra}{Messenger Ribonucleic Acid}
% ------------------------------------ Arrows -----------------------------------------------------------
\draw[line](soma) to (cytoplasm);
\draw[line](soma) to (organelle2);
\draw[line](organelle2) to (nucleus);
\draw[line](cytoplasm) to (organelle1);
\draw[-](organelle1.west) to (mitochondria.east);
\draw[-](organelle1.west) to (golgiApparatus.east);
\draw[-](organelle1.west) to (endoplasmicReticulum.east);
\draw[-](organelle1.west) to (ribosome.east);
\draw[line](endoplasmicReticulum) to (roughEndoplasmicReticulum);
\draw[line](endoplasmicReticulum) to (smoothEndoplasmicReticulum);
\draw[line](nucleus) to (chromosome);
\draw[line](chromosome) to (deoxyribonucleicAcid);
\draw[line](chromosome) to (46chromosomes);
\draw[line](deoxyribonucleicAcid) to (gene);
\draw[line](deoxyribonucleicAcid.east) to (2nm);
\draw[line](deoxyribonucleicAcid.east) to (geneexpression.west);
\draw[arrow, green](gene) to node[above, black, switch ocg={transcription transcriptio}](transcription){
    \basic{transcription}{Transcription}
} (rna);
\node at(transcription){
    \basicr{transcriptio}{?}
};
\draw[arrow,green](rna) to (spliceosome);
\draw[arrow, green](spliceosome) to node[above, black, switch ocg={splicing splicin}](splicing){
    \basic{splicing}{Splicing}} (mrna);
\node at(splicing){
    \basicr{splicin}{?}
};
\draw[arrow, green](mrna) |- (space1.center) to (nucleus);
\draw[arrow, green, out=150, in=30, looseness=0.7, postaction={decorate, decoration={text along path, text align=center, reverse path, raise=5pt, text={Export}}}](pore) to (ribosome);
\draw[arrow, green](ribosome) to (proteinSynthesis);

\question{below right}{4}{transcription}{rnap}{rnapp}{RNA polymerase}
\node[below=of rnap, switch ocg={helicase helicasee}](helicase){
    \basic{helicase}{\textcolor{black}{Using} Helicase}}
node at(helicase){
    \basicr{helicasee}{\textcolor{black}{Using:} ?}
};
\draw[line] (rnap) to (helicase);
\draw[arrow, red!50!green!100!blue!100](transcription) -- (rnap);
\node[default, draw, right=2cm of rnap](dna){DNA};
\draw[arrow, red!50!green!100!blue!100](rnap) to node[above]{Open}(dna);
\node[default, draw, right=2cm of dna](nucleotides){One strand of Nucleotides};
\draw[dashed, brown]($(dna.north)+(0,1)$) -| ($(rnap.east)+(.3,0)$) |- ($(dna.south)+(0,-1)$) -| ($(nucleotides.west)+(-0.3, 0)$) |- node[right, switch ocg={elongation elongationn}](elongation){
        \basic{elongation}{Elongation}
    } cycle;
\node at($(elongation)+(-.5,0)$){
    \basicr{elongationn}{?}
};
\draw[arrow, red!50!green!100!blue!100](dna) to node[above]{Expose}(nucleotides);
\draw[arrow, blue](helicase) -- node[below, switch ocg={guide guidee}](guide){
    \basic{guide}{Guide the nucleotides into position}
}
(nucleotides|-helicase) to (nucleotides);
\node at(guide){\basicr{guidee}{\textcolor{blue}{1. }?}};
\node[below=0 of guide, switch ocg={facilitate facilitatee}](facilitate){
    \basic{facilitate}{Facilitate attachment and elongation}}
node at(facilitate){
    \basicr{facilitatee}{\textcolor{blue}{2. }?}
};
\node[below=0 of facilitate, switch ocg={measure measuree}](measure){
    \basic{measure}{Measure and proofread replacement capability, and termination recognition capability}
}
node(third) at(measure){
    \basicr{measuree}{\textcolor{blue}{3. }?}
};
\draw[arrow, red!50!green!100!blue!100](nucleotides) -- (rna);
\node[default, draw, above=of spliceosome](different){Different mRNAs};
\draw[arrow, gray](spliceosome) to node[right, switch ocg={alternativeSplicing alternativeSplicingg}](alternativeSplicing){
    \basic{alternativeSplicing}{Alternative Splicing}
}
(different);
\node at($(alternativeSplicing)+(-1,0)$){
    \basicr{alternativeSplicingg}{?}
};
\draw[dashed, purple]($(rnap.east)+(.5,0)$) -- +(0,1) -| ($(rnap.west)+(-.5,0)$) --
node[left, switch ocg={transcriptionFactor transcriptionFactorr}](transcriptionFactor){
        \basic{transcriptionFactor}{Transcription Factor}
    }
++(0,-1) -| cycle;
\node at($(transcriptionFactor)+(1.5,0)$){
    \basicr{transcriptionFactorr}{?}
};

% Gene
\node[default, right=8 of nucleotides](gene2){};
\draw(gene2.north) -- +(5,0) |- (gene2.south) -- +(-5,0) node(space2){} |- cycle;
\node[rectangle, draw, left=.17cm of gene2, minimum height=1cm, minimum width=1cm, switch ocg={intron intro}](intron1){
    \basic{intron}{\textcolor{blue}{Intron1}}}
node at(intron1){
    \basicr{intro}{\textcolor{blue}{?}}
};
\node[below=of intron1, switch ocg={splice splicee}](splice){
    \basic{splice}{\textcolor{black}{Removed while} Splicing}}
node at(splice) {
    \basicr{splicee}{\textcolor{black}{Removed while:} ?}
};
\draw[line](intron1) to (splice);
\node[rectangle, draw, switch ocg={pm pmm}, minimum height=1cm, minimum width=.3cm, fill=gray](promotor) at($(gene2)+(-4.7,0)$){};
\node[below=0 of promotor, switch ocg={pm pmm}](promotor2){
    \basic{pm}{Promotor}}
node[below=0 of promotor]{
    \basicr{pmm}{?}
};
\node[above=of promotor, switch ocg={regulatorySequences regulatorySequencess}](regulatorySequences){
    \basic{regulatorySequences}{Regulatory Sequences}}
node at(regulatorySequences){
    \basicr{regulatorySequencess}{\textcolor{black}{Found near it:} ?}
};
\node[minimum width=2cm,minimum height=1cm, switch ocg={exon exo}](exon1)at($(promotor.east)!0.5!(intron1.west)$) {
    \basic{exon}{Exon1}
} node at (exon1){
    \basicr{exo}{?}
};
\node[rectangle, right=0cm of intron1, minimum height=1cm, minimum width=2cm, switch ocg={exon exo}](exon2){
    \basic{exon}{Exon2}
} node at (exon2){
    \basicr{exo}{?}
};
\node[rectangle, draw, right=0cm of exon2, minimum height=1cm, minimum width=2.5cm, switch ocg={intron intro}](intron2){
    \basic{intron}{\textcolor{blue}{Intron2}}}
node at(intron2) {
    \basicr{intro}{\textcolor{blue}{?}}
};
\node[rectangle, draw, switch ocg={term ter}, minimum height=1cm, minimum width=.3cm, fill=gray](terminator)
at($(gene2)+(4.7,0)$){} node[below=0 of terminator, switch ocg={term ter}]{
    \basic{term}{Terminator}}
node[below=0 of terminator]{
    \basicr{ter}{\textcolor{red}{?}}
};
\node[minimum height=1cm, switch ocg={exon exo}](exon2) at($(intron2.east)!0.5!(terminator.west)$){
    \basic{exon}{Exon3}
} node at (exon2){
    \basicr{exo}{?}
};
\draw[line](promotor) to (regulatorySequences);
\node[default, below=2 of gene2](gene3){};
\draw(gene3.north) -- +(4.85,0) |- (gene3.south) -- +(-4.85,0) node(space3){} |- cycle;
\node[draw, rectangle, minimum width=.3cm, minimum height=1cm, fill=blue, switch ocg={cap capp}](cap) at($(gene3)+(-4.7,0)$){};
\node[below=0 of cap, switch ocg={cap capp}](capText){
    \basic{cap}{5' Cap}
}
node at(capText){
    \basicr{capp}{?}
};
\node[draw, rectangle, minimum width=.3cm, minimum height=1cm, fill=green, switch ocg={polya polyaa}](polya) at($(gene3)+(4.7,0)$){};
\node[below=0 of polya, switch ocg={polya polyaa}](polyaText){
    \basic{polya}{3' Poly-A Tail}
}
node at(polyaText){
    \basicr{polyaa}{?}
};
\draw[arrow](gene2) to (gene3);

\node[default, below=2 of gene3, switch ocg={exon exo}](gene4){
    \basic{exon}{Exon}
}
node at(gene4){
    \basicr{exo}{?}
};
\draw(gene4.north) -- +(2.85,0) |- (gene4.south) -- +(-2.85,0) node(space3){} |- cycle;
\node[draw, rectangle, minimum width=.3cm, minimum height=1cm, fill=blue, switch ocg={cap capp}](cap2) at($(gene4)+(-2.7,0)$){};
\node[below=0 of cap2, switch ocg={cap capp}](cap2Text){
    \basic{cap}{5' Cap}
}
node at(cap2Text){
    \basicr{capp}{?}
};
\node[draw, rectangle, minimum width=.3cm, minimum height=1cm, fill=green, switch ocg={polya polyaa}](polya2) at($(gene4)+(2.7,0)$){};
\node[below=0 of polya2, switch ocg={polya polyaa}](polya2Text){
    \basic{polya}{3' Poly-A Tail}
}
node at(polya2Text){
    \basicr{polyaa}{?}
};
\draw[arrow](gene3) to node[right, switch ocg={splicing splicin}](splicing2){
    \basic{splicing}{Splicing}
} (gene4);
\node[switch ocg={splicing splicin}] at($(splicing2)+(-.5,0)$){
    \basicr{splicin}{?}
};
\draw[arrow, dotted](spliceosome) to ($(promotor.west)+(-2pt,0)$);